src/meds_etl/simple_csv.py

```diff
@@ -0,0 +1,220 @@
+                return pl.struct([v.alias(k) for k, v in d.items()])
+
+            metadata = transform_metadata(metadata)
+
+            batch = batch.filter(code.is_not_null())
+
+            event_data = (
+                batch.select(
+                    patient_id=patient_id,
+                    time=time,
+                    code=code,
+                    text_value=text_value,
+                    numeric_value=numeric_value,
+                    datetime_value=datetime_value,
+                    metadata=metadata,
+                    shard=patient_id.hash(213345) % num_shards,
+                )
+                .collect()
+                .partition_by("shard", as_dict=True, maintain_order=False)
+            )
+
+            for shard_index, shard in event_data.items():
+                fname = os.path.join(
+                    temp_dir, str(shard_index), f'{index}_{batch_index}.parquet'
+                )
+                shard.write_parquet(fname, compression="uncompressed")
+
+
+def main():
+    parser = argparse.ArgumentParser(prog="meds_etl_simple_csv", description="Performs an ETL from a simple flat CSV format to MEDS")
+    parser.add_argument("src_csv", type=str)
+    parser.add_argument("destination", type=str)
+    parser.add_argument("--num_shards", type=int, default=100)
+    parser.add_argument("--num_threads", type=int, default=1)
+    args = parser.parse_args()
+
+    if not os.path.exists(args.src_csv):
+        raise ValueError(f'The source CSV folder ("{args.src_csv}") does not seem to exist?')
+
+    os.makedirs(args.destination, exist_ok=True)
+
+    events = []
+
+    decompressed_dir = os.path.join(args.destination, "decompressed")
+    os.mkdir(decompressed_dir)
+
+    temp_dir = os.path.join(args.destination, "temp")
+    os.mkdir(temp_dir)
+
+    for shard_index in range(args.num_shards):
+        os.mkdir(os.path.join(temp_dir, str(shard_index)))
+
+    metadata = {
+        "dataset_name": args.src_csv,
+        "dataset_version": "unknown",
+        "etl_name": "meds_etl.simple_csv",
+        "etl_version": meds_etl.__version__,
+    }
+
+    jsonschema.validate(instance=metadata, schema=meds.dataset_metadata)
+
+    with open(os.path.join(args.destination, "metadata.json"), "w") as f:
+        json.dump(metadata, f)
+
+    all_tasks = []
+
+    random.seed(3422342)
+    random.shuffle(all_tasks)
+
+    for i, csv_file in enumerate(os.listdir(args.src_csv)):
+        full_csv_file = os.path.join(args.src_csv, csv_file)
+        all_tasks.append(
+            (full_csv_file, args.num_shards, temp_dir, decompressed_dir, i)
+        )
+
+    with multiprocessing.get_context("spawn").Pool(args.num_threads) as pool:
+        for _ in pool.imap_unordered(process_file, all_tasks):
+            pass
+
+    shutil.rmtree(decompressed_dir)
+
+    print("Processing each shard")
+
+    data_dir = os.path.join(args.destination, "data")
+    os.mkdir(data_dir)
+
+    for shard_index in range(args.num_shards):
```

Comment:
Why not multiprocess over this too?

Reply:
Multiprocessing over this would require too much RAM